### ISSUE TITLE
Support 16kb page sizes

### ISSUE THREAD

**Issue by mirixh** [firebase/firebase-android-sdk#6101]
## What feature would you like to see?

I would like the sdk to support the 16kb page sizes coming with Android 15 
https://developer.android.com/about/versions/15/behavior-changes-all#16-kb

## How would you use it?

the app would not crash at startup if this was supported

This can be tested with the special emulator image described here https://developer.android.com/guide/practices/page-sizes#16kb-emulator

after doing the suggested changes and launching the app we get the following crash

```
2024-07-15 10:46:41.797  9570-9570  DEBUG                   pid-9570                             A  Cmdline: taxi.android.client.alpha
2024-07-15 10:46:41.797  9570-9570  DEBUG                   pid-9570                             A  pid: 9535, tid: 9535, name: id.client.alpha  >>> taxi.android.client.alpha <<<
2024-07-15 10:46:41.797  9570-9570  DEBUG                   pid-9570                             A        #17 pc 00000000002ee67a  /data/app/~~Q2ksqkOMsGemjDSKkfN5Bw==/taxi.android.client.alpha-fMREGwQV0pgI1iS9epihLA==/base.apk (com.google.firebase.crashlytics.ndk.JniNativeApi.<clinit>+18)
2024-07-15 10:46:41.798  9570-9570  DEBUG                   pid-9570                             A        #23 pc 0000000000d84404  /data/app/~~Q2ksqkOMsGemjDSKkfN5Bw==/taxi.android.client.alpha-fMREGwQV0pgI1iS9epihLA==/oat/arm64/base.odex (com.google.firebase.crashlytics.ndk.a.e+212)
2024-07-15 10:46:41.798  9570-9570  DEBUG                   pid-9570                             A        #24 pc 0000000000b44ebc  /data/app/~~Q2ksqkOMsGemjDSKkfN5Bw==/taxi.android.client.alpha-fMREGwQV0pgI1iS9epihLA==/oat/arm64/base.odex (Nh.a.e+236)
2024-07-15 10:46:41.798  9570-9570  DEBUG                   pid-9570                             A        #25 pc 0000000000b48510  /data/app/~~Q2ksqkOMsGemjDSKkfN5Bw==/taxi.android.client.alpha-fMREGwQV0pgI1iS9epihLA==/oat/arm64/base.odex (O9.d.get+1696)
2024-07-15 10:46:41.798  9570-9570  DEBUG                   pid-9570                             A        #26 pc 0000000000bca4a4  /data/app/~~Q2ksqkOMsGemjDSKkfN5Bw==/taxi.android.client.alpha-fMREGwQV0pgI1iS9epihLA==/oat/arm64/base.odex (V9.l.get+132)
2024-07-15 10:46:41.798  9570-9570  DEBUG                   pid-9570                             A        #27 pc 0000000000bc8864  /data/app/~~Q2ksqkOMsGemjDSKkfN5Bw==/taxi.android.client.alpha-fMREGwQV0pgI1iS9epihLA==/oat/arm64/base.odex (V9.g.h+468)
2024-07-15 10:46:41.798  9570-9570  DEBUG                   pid-9570                             A        #28 pc 0000000000bc8b18  /data/app/~~Q2ksqkOMsGemjDSKkfN5Bw==/taxi.android.client.alpha-fMREGwQV0pgI1iS9epihLA==/oat/arm64/base.odex (V9.g.i+328)
2024-07-15 10:46:41.798  9570-9570  DEBUG                   pid-9570                             A        #29 pc 00000000006584b8  /data/app/~~Q2ksqkOMsGemjDSKkfN5Bw==/taxi.android.client.alpha-fMREGwQV0pgI1iS9epihLA==/oat/arm64/base.odex (O9.h.e+792)
2024-07-15 10:46:41.798  9570-9570  DEBUG                   pid-9570                             A        #30 pc 0000000000657844  /data/app/~~Q2ksqkOMsGemjDSKkfN5Bw==/taxi.android.client.alpha-fMREGwQV0pgI1iS9epihLA==/oat/arm64/base.odex (O9.h.g+1076)
2024-07-15 10:46:41.798  9570-9570  DEBUG                   pid-9570                             A        #31 pc 0000000000657344  /data/app/~~Q2ksqkOMsGemjDSKkfN5Bw==/taxi.android.client.alpha-fMREGwQV0pgI1iS9epihLA==/oat/arm64/base.odex (O9.h.f+244)
2024-07-15 10:46:41.798  9570-9570  DEBUG                   pid-9570                             A        #32 pc 00000000008839bc  /data/app/~~Q2ksqkOMsGemjDSKkfN5Bw==/taxi.android.client.alpha-fMREGwQV0pgI1iS9epihLA==/oat/arm64/base.odex (com.google.firebase.provider.FirebaseInitProvider.onCreate+92)
2024-07-15 10:46:41.798  9570-9570  DEBUG                   pid-9570                             A        #34 pc 0000000000883838  /data/app/~~Q2ksqkOMsGemjDSKkfN5Bw==/taxi.android.client.alpha-fMREGwQV0pgI1iS9epihLA==/oat/arm64/base.odex (com.google.firebase.provider.FirebaseInitProvider.attachInfo+168)
```

running the [validation script ](https://developer.android.com/guide/practices/page-sizes#test)on the final unzipped apk produces the following warnings regarding the crashlytics native libs

```
 ./alignment.sh /tmp/apk_out |grep "arm64-v8a"
/tmp/apk_out/lib/arm64-v8a/libcrashlytics-common.so: \e[31mUNALIGNED\e[0m (2**12)
/tmp/apk_out/lib/arm64-v8a/libcrashlytics.so: \e[31mUNALIGNED\e[0m (2**12)
/tmp/apk_out/lib/arm64-v8a/libcrashlytics-handler.so: \e[31mUNALIGNED\e[0m (2**12)
/tmp/apk_out/lib/arm64-v8a/libcrashlytics-trampoline.so: \e[31mUNALIGNED\e[0m (2**12)
```

**Comment by lehcar09** [comment]
Hi @mirixh, thank you for reaching out. We have released an updated Crashlytics NDK [version 19.0.2](https://firebase.google.com/support/release-notes/android?hl=en#crashlytics-ndk_v19-0-2) last Jun 20, 2024 that supports 16Kb page sizes. 

With that, I'll be closing this now. If you encounter an issue, please file a new ticket with the details that can help us investigate the issue. Thanks!

**Comment by mirixh** [comment]
thanks for fast feedback and sorry for the inconvenience, 

had lost track of this
